test/integration/evictions/evictions_test.go

```diff
@@ -0,0 +1,242 @@
+
+	rm := disruption.NewDisruptionController(
+		informers.Pods().Informer(),
+		clientset.NewForConfigOrDie(restclient.AddUserAgent(&config, "replicaset-controller")),
+	)
+	if err != nil {
+		t.Fatalf("Failed to create replicaset controller")
+	}
+	return s, rm, informers.Pods().Informer(), clientSet
+}
+
+func TestConcurrentEvictionRequests(t *testing.T) {
+	podNameFormat := "test-pod-%d"
+
+	s, rm, podInformer, clientSet := rmSetup(t)
+	defer s.Close()
+
+	ns := framework.CreateTestingNamespace("concurrent-eviction-requests", s, t)
+	defer framework.DeleteTestingNamespace(ns, s, t)
+
+	stopCh := make(chan struct{})
+	go podInformer.Run(stopCh)
+	go rm.Run(stopCh)
+
+	config := restclient.Config{Host: s.URL}
+	clientSet, err := clientset.NewForConfig(&config)
+
+	var gracePeriodSeconds int64 = 30
+	deleteOption := &v1.DeleteOptions{
+		GracePeriodSeconds: &gracePeriodSeconds,
+	}
+
+	// Create 10 pods to evict
+	for i := 0; i < 10; i++ {
+		pod := &v1.Pod{
+			ObjectMeta: v1.ObjectMeta{
+				Name:   fmt.Sprintf(podNameFormat, i),
+				Labels: map[string]string{"app": "test-evictions"},
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:  "fake-name",
+						Image: "fakeimage",
+					},
+				},
+			},
+			Status: v1.PodStatus{
+				Phase: v1.PodRunning,
+				Conditions: []v1.PodCondition{
+					{
+						Type:   v1.PodReady,
+						Status: v1.ConditionTrue,
+					},
+				},
+			},
+		}
+
+		if _, err := clientSet.Core().Pods(ns.Name).Create(pod); err != nil {
+			t.Errorf("Failed to create pod: %v", err)
+		}
+
+		clientSet.Core().Pods(ns.Name).UpdateStatus(pod)
+	}
+
+	waitToObservePods(t, podInformer, 10)
+
+	pdb := &v1beta1.PodDisruptionBudget{
+		ObjectMeta: v1.ObjectMeta{
+			Name: fmt.Sprintf("test-pdb"),
+		},
+		Spec: v1beta1.PodDisruptionBudgetSpec{
+			MinAvailable: intstr.IntOrString{
+				Type:   intstr.Int,
+				IntVal: 0,
+			},
+			Selector: &unversioned.LabelSelector{
+				MatchLabels: map[string]string{"app": "test-evictions"},
+			},
+		},
+	}
+	if _, err := clientSet.Policy().PodDisruptionBudgets(ns.Name).Create(pdb); err != nil {
+		t.Errorf("Failed to create PodDisruptionBudget: %v", err)
+	}
+
+	waitPDBStable(t, clientSet, 10, ns.Name, pdb.Name)
+
+	doneCh := make(chan bool, 10)
+	errCh := make(chan error, 1)
+	// spawn 10 goroutine to concurrently evict the pods
+	for i := 0; i < 10; i++ {
+		go func(id int, doneCh chan bool, errCh chan error) {
+			eviction := &v1beta1.Eviction{
+				TypeMeta: unversioned.TypeMeta{
+					APIVersion: "Policy/v1beta1",
+					Kind:       "Eviction",
+				},
+				ObjectMeta: v1.ObjectMeta{
+					Name:      fmt.Sprintf(podNameFormat, id),
+					Namespace: ns.Name,
+				},
+				DeleteOptions: deleteOption,
+			}
+
+			var e error
+			for {
+				e = clientSet.Policy().Evictions(ns.Name).Evict(eviction)
+				if errors.IsTooManyRequests(e) {
+					time.Sleep(5 * time.Second)
+				} else {
+					break
+				}
+			}
+			if e != nil {
+				if errors.IsConflict(err) {
```

Comment:
The message should make it clear that we should never get a 409 in such a case. As is, it doesn't add much value.